src/core/server/environment/environment_service.ts

```diff
@@ -40,7 +40,9 @@ export class EnvironmentService {
     this.configService = core.configService;
   }
 
-  public async setup() {
+  public async preboot() {
+    // IMPORTANT: This code is based on the assumption that none of the configuration values used
+    // here is supposed to change during preboot phase and it's safe to read them only once.
```

Comment:
I tried to avoid making changes that we may never need, but it wouldn't be difficult to account for the dynamic changes in config here if we ever need this.